packages/cli/api-importers/v2-importer-commons/src/converters/schema/SchemaConverter.ts

```diff
@@ -173,11 +193,209 @@ export class SchemaConverter extends AbstractConverter<AbstractConverterContext<
+                                v1: "UUID",
+                                v2: FernIr.PrimitiveTypeV2.uuid({})
+                            });
+                        case "BIG_INTEGER":
+                            return FernIr.TypeReference.primitive({
+                                v1: "BIG_INTEGER",
+                                v2: FernIr.PrimitiveTypeV2.bigInteger({
+                                    default: undefined
+                                })
+                            });
+                        default:
+                            return undefined;
+                    }
+                },
+                unknown: () => {
+                    return FernIr.TypeReference.unknown();
+                },
+                map: ({ keyType, valueType }) => {
+                    if (keyType == null || valueType == null) {
+                        return undefined;
+                    }
+                    return FernIr.TypeReference.container(
+                        FernIr.ContainerType.map({
+                            keyType,
+                            valueType
+                        })
+                    );
+                },
+                list: (itemType) => {
+                    if (itemType == null) {
+                        return undefined;
+                    }
+                    return FernIr.TypeReference.container(FernIr.ContainerType.list(itemType));
+                },
+                optional: (itemType) => {
+                    if (itemType == null) {
+                        return undefined;
+                    }
+                    return FernIr.TypeReference.container(FernIr.ContainerType.optional(itemType));
+                },
+                nullable: (itemType) => {
+                    if (itemType == null) {
+                        return undefined;
+                    }
+                    return FernIr.TypeReference.container(FernIr.ContainerType.nullable(itemType));
+                },
+                set: (itemType) => {
+                    if (itemType == null) {
+                        return undefined;
+                    }
+                    return FernIr.TypeReference.container(FernIr.ContainerType.set(itemType));
+                },
+                literal: (literal) => {
+                    return FernIr.TypeReference.container(
+                        FernIr.ContainerType.literal(
+                            literal._visit<FernIr.Literal>({
+                                string: (value) => FernIr.Literal.string(value),
+                                boolean: (value) => FernIr.Literal.boolean(value),
+                                _other: () => {
+                                    throw new Error("Unexpected literal type");
+                                }
+                            })
+                        )
+                    );
+                },
+                named: () => {
+                    return undefined;
```

Comment:
Discussed offline - worth noting that `x-fern-type` doesn't support named types (hence why we return `undefined` here). We would use `x-fern-type-name` for that.